client/internal/receivedprocessor.go

```diff
@@ -57,26 +62,63 @@ func (r *receivedProcessor) ProcessReceivedMessage(ctx context.Context, msg *pro
 			r.logger.Errorf("cannot processed received flags:%v", err)
 		}
 
-		msgData := &types.MessageData{
-			RemoteConfig: msg.RemoteConfig,
+		msgData := &types.MessageData{}
+
+		if msg.RemoteConfig != nil {
+			if r.capabilities&protobufs.AgentCapabilities_AcceptsRemoteConfig != 0 {
```

Comment:
This is really minor, but I find these hard to read especially with the decision in go to omit space on either side of the ampersand. I would refactor to a function:

```go
func (r *receivedProcessor) hasCapability(capability AgentCapabilities) {
    return r.capabilities&capability != 0
}
```

And then use that throughout.

Reply:
I like it. Done.